polkadot/xcm/xcm-executor/src/traits/asset_exchange.rs

```diff
@@ -37,6 +37,27 @@ pub trait AssetExchange {
 		want: &Assets,
 		maximal: bool,
 	) -> Result<AssetsInHolding, AssetsInHolding>;
+
+	/// Handler for quoting the exchange price of two asset collections.
+	///
+	/// It's useful before calling `exchange_asset`, to get some information on whether or not the
+	/// exchange will be successful.
+	///
+	/// Arguments:
+	/// - `give` The asset(s) that are going to be given.
+	/// - `want` The asset(s) that are wanted.
+	/// - `maximal`:
+	/// 	  - If `true`, then the return value is the resulting amount of `want` obtained by swapping
+	///      `give`.
+	///   - If `false`, then the return value is the required amount of `give` needed to get `want`.
+	///
+	/// The return value is `Assets` since it comprises both which assets and how much of them.
+	///
+	/// The relationship between this function and `exchange_asset` is the following:
+	/// - quote(give, want, maximal) = resulting_want -> exchange(give, resulting_want, maximal) ✅
+	/// - quote(give, want, minimal) = required_give -> exchange(required_give_amount, want,
```

Comment:
this useful explanation. 
would not this be simpler if we do something like:
```
quote_exchange(give: Assets, want: AssetIds) -> Option<Assets>;
quote_price(give: AssetIds, want: Assets) -> Option<Assets>;
```

Reply:
Hmm, it makes more sense to use `AssetId` instead of the assets and the amount whenever we don't need the amount.

However, there are two things I like about the API as-is.
One, by specifying the fungibility we get to also specify instances for NFTs.
Two, the addition of `maximal` even though it's weird it matches better with `exchange_asset`, so you know which one matches with which one.